File: 49-group-anagrams/49-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, ArrayList> m = new HashMap<>();
        for (String str : strs) {
            int[] count = new int[26];
            for (char c : str.toCharArray()) {
                count[c-'a']++;
            }
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < 26; i++) {
                sb.append('.');
                sb.append(count[i]);
            }
            String key = sb.toString();
            if (!m.containsKey(key)) {
                m.put(key, new ArrayList<>());
            }
            m.get(key).add(str);
        }
        
        return new ArrayList(m.values());
    }
}

File: 49-group-anagrams/49-group-anagrams.cpp

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> m;
        for (int i = 0; i < strs.size(); i++) {
            vector<int> chars(26);
            for (int j = 0; j < strs[i].size(); j++) {
                chars[strs[i][j]-'a']++;
            }
            string key = "";
            for (int j = 0; j < 26; j++) {
                key.append(to_string(chars[j]+'a'));
            }
            
            m[key].push_back(strs[i]);
        }
                           
        vector<vector<string>> res;
        for (auto i = m.begin(); i != m.end(); i++) {
            res.push_back(i->second);
        }
        return res;
    }
};